proto/src/determined/api/v1/experiment.proto

```diff
@@ -297,6 +307,9 @@ message MetricBatchesResponse {
 
 // Request metrics from all trials at a progress point of progress.
 message TrialsSnapshotRequest {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "experiment_id", "batches_processed", "metric_name", "metric_type" ] }
+  };
   // The id of the experiment.
   int32 experiment_id = 1;
   // The point of progress at which to query metrics.
```

Comment:
on the line below
`int32 batches_processed = 2 [(grpc.gateway.protoc_gen_swagger.options.openapiv2_field) = {required: ['batches_processed']}];`
seems to produce the expected result. found it just ignoring what we're doing and trying to figure out how to do it independently.

Reply:
I see what we're doing documented though, not sure why it doesn't work.

Reply:
https://github.com/grpc-ecosystem/grpc-gateway/issues/665 ok that explains it.

Reply:
Yep that appears to be working to me. Going with that!